controller_manager:
  ros__parameters:
    update_rate: 1000

    joint_state_broadcaster:
      type: joint_state_broadcaster/JointStateBroadcaster

    trajectory_controller:
      type: joint_trajectory_controller/JointTrajectoryController

trajectory_controller:
  ros__parameters:
    joints:
      # LEFT HIP
      - left_hip_roll_joint
      - left_hip_pitch_joint
      - left_hip_yaw_joint
      # LEFT KNEE
      - left_knee_roll_joint
      - left_knee_pitch_joint
      - left_knee_yaw_joint
      # LEFT ANKLE
      - left_ankle_roll_joint
      - left_ankle_pitch_joint
      - left_ankle_yaw_joint

      # RIGHT HIP
      - right_hip_roll_joint
      - right_hip_pitch_joint
      - right_hip_yaw_joint
      # RIGHT KNEE
      - right_knee_roll_joint
      - right_knee_pitch_joint
      - right_knee_yaw_joint
      # RIGHT ANKLE
      - right_ankle_roll_joint
      - right_ankle_pitch_joint
      - right_ankle_yaw_joint

    command_interfaces:
      - position
    state_interfaces:
      - position
      - velocity

    action_monitor_rate: 100.0
    allow_partial_joints_goal: false
    allow_nonzero_velocity_at_trajectory_end: false

    constraints:
      stopped_velocity_tolerance: 0.05
      goal_time: 0.0

      # You can keep per-joint tolerances if you want
      left_hip_roll_joint:   {trajectory: 0.05, goal: 0.02}
      left_hip_pitch_joint:  {trajectory: 0.05, goal: 0.02}
      left_hip_yaw_joint:    {trajectory: 0.05, goal: 0.02}

      left_knee_roll_joint:  {trajectory: 0.05, goal: 0.02}
      left_knee_pitch_joint: {trajectory: 0.05, goal: 0.02}
      left_knee_yaw_joint:   {trajectory: 0.05, goal: 0.02}

      left_ankle_roll_joint:  {trajectory: 0.05, goal: 0.02}
      left_ankle_pitch_joint: {trajectory: 0.05, goal: 0.02}
      left_ankle_yaw_joint:   {trajectory: 0.05, goal: 0.02}

      right_hip_roll_joint:   {trajectory: 0.05, goal: 0.02}
      right_hip_pitch_joint:  {trajectory: 0.05, goal: 0.02}
      right_hip_yaw_joint:    {trajectory: 0.05, goal: 0.02}

      right_knee_roll_joint:  {trajectory: 0.05, goal: 0.02}
      right_knee_pitch_joint: {trajectory: 0.05, goal: 0.02}
      right_knee_yaw_joint:   {trajectory: 0.05, goal: 0.02}

      right_ankle_roll_joint:  {trajectory: 0.05, goal: 0.02}
      right_ankle_pitch_joint: {trajectory: 0.05, goal: 0.02}
      right_ankle_yaw_joint:   {trajectory: 0.05, goal: 0.02}
